sfcasts/ep1/mocking-mock-objects.md

```diff
@@ -1,94 +1,97 @@
-ensure that things are working as expected, let's go ahead and use it in our
-controller. So open up the main controller and here at the public function index,
-
-Let's go ahead and pass in a GitHub service argument and we'll call it GitHub. Thanks
-to Symfony the service, be configured automatically and inject it into this method.
-Anytime it's called right below our dinos array, we're going to create a for each
-loop and for each dinos as Dino, we want to set the health of the Dino by doing Dino
-set health and to set the health, we're going to call our GitHub service, get health
-report. And of course we have to pass in the name of our Dino to the GI health report
-method. And that's it moving back to the browser refresh and oh, no, the return value
-must be of type health status and NOLA was returned. Hmm. Let's take a look back at
-our code and we'll look here, uh, our get health report method. Pause.
-
-All right. So we move back to our browser. We refresh and oh, no. All right. So get,
-get Dyna status from labels. Return value must be of type health status and no is
-returned later on. We can actually write a test for this, uh, in a future episode,
-but for right now will, let's go back and we can see here, uh, for our type area,
-that line 54 return health status try from status is returning Noll. I bet this has
-something to do with, uh, an issue in GitHub. Let me look at the issues real quick.
-And we have, huh, Dennis, Dennis just finished his daily exercise routine and he is
-hungry. The reason why we're getting this error is if we go and look in our GitHub
-service, we'll scroll down to the GI dinosaur status from labels method. And we can
-see that hungry, uh, that we're passing the sta status from GitHub to our health
-status, try from method. This is our enum. And if we look in our health status, uh,
-enum, we don't have a case for hungry. Ha go figure coming up next. We're going to
-figure how to, uh, we're going to figure that one out.
+Our tests are passing, the dino's are wandering, and life is great! But... let's
+think about this for a second. In `GithubService`, when we test `getHealthReport()`,
+we're able to control the `$response` that we get back from `request()` by using
+a stub. But we also need to ensure that the service is only calling GitHub one time,
+*and* it's using the right HTTP method with the correct URL. How do we do that?
+
+In `GithubServiceTest` where we configure the `$mockHttpClient`, add `->expects()`,
+and pass in the `once()` `InvocationRule` needed for this method.
+
+Over in the terminal, run our tests...
+
+```terminal
+./vendor/bin/phpunit
+```
+
+And... Awesome! We've just added an assertion to our mock client that requires the
+`request` method to be called *exactly* one time. Let's take it a step further and
+add `->with()` to the `$mockHttpClient`. For the arguments, pass in `GET` and the
+URL for GitHub: `https://api.github.com/repos/SymfonyCasts/dino-park/`.
```

Comment:
Hmmm... im not going to say the actual url in the video..  Think this should be
> For the arguments, pass in `GET` and then I'll copy the URL for GitHub...

Reply:
> For the arguments, pass in GET and then I'll paste the URL we need to the GitHub API.